src/main/java/com/arangodb/internal/ArangoCollectionImpl.java

```diff
@@ -105,6 +112,32 @@ public DocumentImportEntity importDocuments(final Collection<?> values, final Do
 		return executor.execute(importDocumentsRequest(values, options), DocumentImportEntity.class);
 	}
 
+	@Override
+	public Collection<DocumentImportEntity> importDocuments(Collection<?> values, DocumentImportOptions options,
+															int batchSize, int numThreads) throws ArangoDBException {
+		List<? extends List<?>> batches = ListUtils.partition(new ArrayList<>(values), batchSize);
+		ExecutorService executorService = Executors.newFixedThreadPool(numThreads);
+		List<CompletableFuture<DocumentImportEntity>> completableFutureList = new ArrayList<>();
+		for (List<?> batch : batches) {
+			CompletableFuture<DocumentImportEntity> completableFuture = CompletableFuture.supplyAsync(() -> {
+				DocumentImportEntity documentImportEntity = importDocuments(batch, options);
+				return documentImportEntity;
+			}, executorService);
+			completableFutureList.add(completableFuture);
+		}
+		List<DocumentImportEntity> documentImportEntityList = new ArrayList<>();
+		for (CompletableFuture<DocumentImportEntity> completableFuture : completableFutureList) {
+			DocumentImportEntity documentImportEntity = null;
+			try {
+				documentImportEntity = completableFuture.get();
+			} catch (InterruptedException | ExecutionException e) {
+				throw new ArangoDBException(e);
+			}
+			documentImportEntityList.add(documentImportEntity);
+		}
```

Comment:
@hkernbach I noticed a small problem.  
RIght before the return statement on this [line](https://github.com/arangodb/arangodb-java-driver/blob/master/src/main/java/com/arangodb/internal/ArangoCollectionImpl.java#L138l)
we need to shutdown the executorService.
```
        executorService.shutdown();
```
If another pull request is needed, please let me know.

Reply:
Will fix that, thanks for spotting :)